src/appmixer/utils/controls/OnError/OnError.js

```diff
@@ -62,28 +57,22 @@ module.exports = {
         try {
             lock = await context.lock(lockName, { ttl: context.config.lockTTL || 1000 * 60 * 5, maxRetryCount: 0 });
             const { lastLogId, gridTimestamp } = context.state;
-            // ensure that if there is a limit set by context.config.limit
             const limit = context.properties.limit
-                ? Math.min(
-                    parseInt(context.properties.limit),
-                    parseInt(context.config.limit) || Infinity
-                )
-                : parseInt(context.config.limit) || 100;
-
-            const result = await fetchErrorsPage(context, lock, limit, lastLogId, gridTimestamp);
-
-            if (result?.length > 0) {
-                // Process and send the errors to outport
-                const labeledErrors = addLabels(context, result);
-                await context.sendArray(labeledErrors, 'out');
-
-                // Update state with the _id of the last log in the page
-                const newLastLogId = result[result.length - 1]['_id'];
-                const newGridTimestamp = result[result.length - 1]['gridTimestamp'];
-                return context.saveState({ lastLogId: newLastLogId, gridTimestamp: newGridTimestamp });
+                ? Math.min(parseInt(context.properties.limit), parseInt(context.config.limit) || Infinity)
+                : parseInt(context.config.limit) || 1000;
+
+            for await (const batch of fetchErrorsGenerator(context, lock, limit, lastLogId, gridTimestamp)) {
+                if (batch.length > 0) {
+                    // Update state with the _id of the last log in the page
+                    const newLastLogId = batch[batch.length - 1]['_id'];
+                    const newGridTimestamp = batch[batch.length - 1]['gridTimestamp'];
+                    await context.saveState({ lastLogId: newLastLogId, gridTimestamp: newGridTimestamp });
```

Comment:
Always send a message to the output port and then save the `state`. In case of a crash after this line. The `tick` will be generated again, but it will read the `state` and one batch could be lost. In case of a crash, we rather duplicate data on the output port, rather than skip something.